### PR TITLE
purge synclogs on form submission

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_sync_purge.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import uuid
+
+from couchdbkit.exceptions import ResourceNotFound
+from django.test import TestCase
+
+from casexml.apps.case.tests.util import delete_all_xforms, delete_all_sync_logs
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/test_sync_purge.py]: F401 'delete_all_sync_logs' imported but unused
F401 'delete_all_xforms' imported but unused

File: corehq/ex-submodules/casexml/apps/case/tests/test_sync_purge.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from casexml.apps.phone.models import get_properly_wrapped_sync_log
+from casexml.apps.phone.tests.utils import create_restore_user, generate_restore_payload, \
+    synclog_id_from_restore_payload, generate_restore_response
+from corehq.apps.domain.models import Domain
+from corehq.apps.receiverwrapper.util import submit_form_locally
+from corehq.apps.users.models import CommCareUser
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/test_sync_purge.py]: F401 'CommCareUser' imported but unused

File: corehq/ex-submodules/casexml/apps/case/util.py [hunk 3]
```diff
@@ -134,6 +135,17 @@ def update_sync_log_with_checks(sync_log, xform, cases, case_db,
                                         case_id_blacklist=case_id_blacklist)
 
 
+def prune_previous_log(sync_log):
```
Comment (emord) [inline corehq/ex-submodules/casexml/apps/case/util.py]: This adds three queries to couch for every form submission. Should we attempt to batch these and perform deletions asynchronously?

Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/util.py]: Only for the first submission after a sync

Comment (emord) [inline corehq/ex-submodules/casexml/apps/case/util.py]: I think that means every submission for icds as they are trained to sync after every form.

Anyways not necessary to fix this. Looks like tests are failing in a weird way though (happened twice in a row)

**Pull request merged**